### PR TITLE
Add forms for the like,comment and post

### PR DESCRIPTION
**Pull request by iankiema**
**Key Changes:**

- Added a new post creation form to the PostsController#new view.
- Integrated a comment form within the PostsController#show view for easy comment addition.
- Ensured proper routing and controller actions for handling form submissions.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -7,10 +7,22 @@
...
-  
+
+    <%= form_for [@post, Comment.new] do |f| %>
+        <div class="field">
+             <%= f.label :text %>
+          <%= f.text_area :text %>
+        </div>
+
+         <%= f.submit "Add Comment" %>
+    <% end %>
```
Comment (Strangeal) [inline app/views/posts/show.html.erb]: - [ ] Great progress on the form implementation! 👏 Your effort is evident. I'd like to suggest a slight improvement in the structure of your routes, specifically for the **comment form**.

   - Consider creating a dedicated route for the comment form, such as `/posts/:post_id/comments/new`. Currently, you're rendering the form on `/users/:user_id/posts/:id`, which is intended for viewing post details. Separating the comment form onto its own route enhances the organization and clarity of your application.

   - After successfully submitting a comment, it's good practice to redirect users to the post details page (`/users/:user_id/posts/:id`). This provides a seamless user experience, allowing them to  see their comment in the context of the post. See below for more details on your UI:
Screenshot 2023-11-10 at 10 45 58

   

Making these adjustments ensures a cleaner structure and a more intuitive user flow. 
Happy doing! 😊🚀

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<!-- app/views/posts/new.html.erb -->
+<h1>New Post</h1>
+
+<%= render 'form', post: @post %>
+
+<%= link_to 'Back', posts_path %>
```
Comment (Strangeal) [inline app/views/posts/new.html.erb]: - [ ] Great progress on the form implementation! 👏 Your effort is evident. I'd like to suggest a slight improvement in the structure of your routes, specifically for the **posts form**.

    - Consider creating a dedicated route for the posts form, such as `/posts/new`. Currently, you're rendering the form on `/users/:user_id/`, which is intended for viewing user details. Separating the post form onto its own route enhances the organization and clarity of your application.

    - After successfully submitting a post, it's good practice to redirect users to the user details page (`/users/:user_id/`). This provides a seamless user experience, allowing them to see their posts in the context of the post. See below for more details on your UI:
Screenshot 2023-11-10 at 10 45 20


Making these adjustments ensures a cleaner structure and a more intuitive user flow. 
Happy coding! 😊🚀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @iankiema, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @iankiema, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well fixing the app routes for the comments form.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**